drivers/i2c/i2c_ll_stm32.h

```diff
@@ -11,13 +11,31 @@
 
 typedef void (*irq_config_func_t)(struct device *port);
 
+#ifdef CONFIG_I2C_STM32_V2
+/**
+ * @brief structure to convey optional i2c timings settings
+ */
+struct i2c_config_timing {
+	/* i2c periphral clock in Hz */
+	uint32_t periph_clock;
+	/* i2c bus speed in Hz */
+	uint32_t i2c_speed;
+	/* I2C_TIMINGR register value of i2c v2 peripheral */
+	uint32_t timing_setting;
```

Comment:
maybe just name `timingr`?

Reply:
I know that this is the register name but I just felt that timingr is not meaningful in the code and structure  .... though if everyone agrees, I can do so

Reply:
I don't have a string opinion here, @pabigot maybe ?

Reply:
I don't have standing to make the decision, just raise the concern.  By my process if the SOC maintainer @erwango defers, then it defers to the implementer @LMESTM.  I'm fine with `timing_setting`.